zipline/pipeline/loaders/earnings_estimates.py

```diff
@@ -318,35 +318,38 @@ def collect_overwrites_for_sid(self,
         sid : int
             The sid for which overwrites should be computed.
         """
-        next_qtr_start_indices = dates.searchsorted(
-            group[EVENT_DATE_FIELD_NAME].values,
-            side=self.searchsorted_side,
-        )
+        # If data was requested for only 1 date, there can never be any
+        # overwrites, so skip the extra work.
+        if len(dates) > 1:
```

Comment:
We could also do:

```
# Assuming `dates` of `0` is never passed her.
if len(dates) == 1:
    return
```

or:
```
if len(dates) <= 1:
    return
```

Which would remove the need for the extra level indentation.
I feel the suggested version is more explicit about bailing/short-circuiting on the case of only one day, but could either way.